go.mod

```diff
@@ -1,8 +1,6 @@
 module github.com/vmware-tanzu/velero
 
-go 1.22.0
-
-toolchain go1.22.11
+go 1.23.0
```

Comment:
This is required by
1. https://go-review.googlesource.com/c/crypto/+/652135 is only included in v0.35.0 which has minimum go directive of 1.23 from https://go.googlesource.com/crypto/+/89ff08d67c4d79f9ac619aaf1f7388888798651f due to 1.22 EOL.

Reply:
why start w/ 1.23.0 latest 1.23 is 1.23.6 

go1.23.5 (released 2025-01-16) includes security fixes to the crypto/x509 and net/http packages, as well as bug fixes to the compiler, the runtime, and the net package. See the [Go 1.23.5 milestone](https://github.com/golang/go/issues?q=milestone%3AGo1.23.5+label%3ACherryPickApproved) on our issue tracker for details.

go1.23.6 (released 2025-02-04) includes security fixes to the crypto/elliptic package, as well as bug fixes to the compiler and the go command. See the [Go 1.23.6 milestone](https://github.com/golang/go/issues?q=milestone%3AGo1.23.6+label%3ACherryPickApproved) on our issue tracker for details.

Reply:
This is go directive. It does not dictate the go toolchain version used to build. It should not be "as high as possible". This value should not be used to fixed CVEs.

#8397 

As seen in https://go.googlesource.com/crypto/+/89ff08d67c4d79f9ac619aaf1f7388888798651f
The golang gopher robot set it to minimum supported version, not highest possible version.

Reply:
Until https://github.com/actions/setup-go/issues/481 looks like toolchain is needed as we are using [`go-version-file: go.mod`](https://github.com/search?q=repo%3Avmware-tanzu%2Fvelero+%22go-version-file%3A+%27go.mod%27%22&type=code)